clients/src/main/resources/message/CreatePartitionsResponse.json

```
@@ -0,0 +1,35 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+{
+  "apiKey": 37,
+  "type": "response",
+  "name": "CreatePartitionsResponse",
+  // Starting in version 1, on quota violation, brokers send out responses before throttling.
+  "validVersions": "0-1",
+  "fields": [
+    { "name": "ThrottleTimeMs", "type": "int32", "versions": "0+",
+      "about": "The duration in milliseconds for which the request was throttled due to a quota violation, or zero if the request did not violate any quota." },
+    { "name": "Results", "type": "[]CreatePartitionsTopicResult", "versions": "0+",
+      "about": "The partition creation results for each topic.", "fields": [
+      { "name": "Name", "type": "string", "versions": "0+",
+        "about": "The topic name." },
+      { "name": "ErrorCode", "type": "int16", "versions": "0+",
```

We have found it useful to document the possible error types inside the `XResponse` classes. Do you think that is suitable meta information to be included in the json schema somewhere?

---

I guess the question is what you would expect to do with that information.  I guess you could throw an exception if someone tried to send an error that wasn't allowed by a given version.  That would be a good sanity check on the server side.

Probably the most difficult thing would be expressing the concept of versioning here in a readable way.  Having a separate entry each time a new allowable error is added (or removed) feels very unreadable.  We could perhaps have a map from values to versions like:
```
{
"0,1": "0+",
"2": "1+"
}
```
The other thing which we would probably want here is the ability to give names to the 0, 1, 2, etc. constants.  This is all doable but just would require a lot of work.

So, to make a long story short, I think we should consider this for the future.  In the meantime we are documenting this in comments in the JSON file.

---

Primarily it is for documentation. Occasionally we will even bump a protocol version just to be able to return a new error code safely even if the schema is unchanged otherwise. It is useful to track this information _somewhere_ and I can't think of a better place than the schema.

---

Sounds good.  I created a follow-up JIRA here: https://issues.apache.org/jira/browse/KAFKA-7787